### PR TITLE
feature(datacatalog): add datacatalog policy tag manager snippets

### PR DESCRIPTION
**Pull request by shollyman**
This PR adds examples of interacting with the policytag manager client within datacatalog, which is leveraged by BigQuery
for defining column level ACLs.

DO NOT MERGE:  currently relies on v1beta1, needs to migrate to v1 before publication

### CODE REVIEW

File: datacatalog/snippets/policytagmanager/create_policytag.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	datacatalog "cloud.google.com/go/datacatalog/apiv1beta1"
+	datacatalogpb "google.golang.org/genproto/googleapis/cloud/datacatalog/v1beta1"
+)
+
+// createPolicyTag creates a policy tag resource under a given parent taxonomy.
+//
+// It optionally accepts a parent ID, which can be used to create a hierarchical
+// relationship between tags.
+func createPolicyTag(parent, displayName, parentPolicyTag string, w io.Writer) (string, error) {
```
Comment (tbpg) [inline datacatalog/snippets/policytagmanager/create_policytag.go:31]: All of our other samples put `w` or `ctx, w` first. Might switching this?

File: datacatalog/snippets/policytagmanager/get_iam_policy.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	"io"
+	"strings"
+
+	datacatalog "cloud.google.com/go/datacatalog/apiv1beta1"
+	iampb "google.golang.org/genproto/googleapis/iam/v1"
+)
+
+// getIAMPolicy prints information about the IAM policy associated with a
+// given taxonomy or policy tag resource.
+func getIamPolicy(resourceID string, w io.Writer) error {
```
Comment (tbpg) [inline datacatalog/snippets/policytagmanager/get_iam_policy.go:30]: s/Iam/IAM/

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

**Review by tbpg — CHANGES_REQUESTED** [review]
Sorry for the slow review! Fell off my radar. Just a couple small comments.

File: datacatalog/snippets/policytagmanager/create_policytag.go (hunk 1 above)
Comment (shollyman) [reply datacatalog/snippets/policytagmanager/create_policytag.go:31]: done

File: datacatalog/snippets/policytagmanager/get_iam_policy.go (hunk 2 above)
Comment (shollyman) [reply datacatalog/snippets/policytagmanager/get_iam_policy.go:30]: File this one under "I spend too much time working with discovery libraries."

Done.

**Review by tbpg — APPROVED** [review]

**Pull request merged**